File: google/cc/as/plus_one.cc
#include <bits/stdc++.h>
typedef long long ll;
using namespace std;

class Solution {
  public:
    vector<int> plusOne(vector<int> &digits) {
        vector<int> ans;
        int keta = 1;
        for(int i = (int)digits.size() - 1; i >= 0; i--) {
            int res = digits[i] + keta;
            if(res > 9) {
                ans.push_back(res % 10);
                keta = 1;
            } else {
                ans.push_back(res);
                keta = 0;
            }
        }
		if(keta)
			ans.push_back(keta);
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: src/pick_one/plus_one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        result = list(digits)
        n = len(result)
        for index in range(n)[::-1]:
            if result[index] == 9:
                result[index] = 0
                if index == 0:
                    return [1] + result
            else:
                result[index] += 1
                return result
